disco/cli/make_upgrade_tables.py

```diff
@@ -0,0 +1,150 @@
+
+    serialize_table(upgrade_summary_table, output_path / "upgrade_summary.csv")
+    serialize_table(total_upgrade_costs_table, output_path / "total_upgrade_costs.csv")
+
+
+def parse_job_results(job, output_path):
+    job_path = output_path / JOBS_OUTPUT_DIR / job.name
+    deployment = job.model.deployment
+    job_info = JobInfo(
+        name=job.name,
+        substation=deployment.substation,
+        feeder=deployment.feeder,
+        placement=deployment.project_data["placement"],
+        sample=deployment.project_data["sample"],
+        penetration_level=deployment.project_data["penetration_level"],
+    )
+    upgrade_summary_table = get_upgrade_summary_table(job_path, job_info)
+    total_upgrade_costs_table = get_total_upgrade_costs_table(job_path, job_info)
+    
+    return (
+        upgrade_summary_table,
+        total_upgrade_costs_table
+    )
+
+
+def get_upgrade_summary_table(job_path, job_info):
+
+    def _get_upgrade_summary(upgrade_type, upgrade_summary_file):
+        if not upgrade_summary_file.exists():
+            return []
+        
+        try:
+            df = pd.read_csv(upgrade_summary_file)
+        except pd.errors.EmptyDataError as e:
+            logger.error("Failed to parse upgrade summary file - '%s'", upgrade_summary_file)
+            logger.exception(e)
+            return []
+        
+        summary_result = []
+        records = df.to_dict("records")
+        for record in records:
+            data = job_info._asdict()
+            data.update({"upgrade_type": upgrade_type})
+            data.update(record)
+            summary_result.append(data)
+        return summary_result
+
+    upgrade_summary = []
+
+    thermal_upgrade_summary_file = job_path / "ThermalUpgrades" / "thermal_summary.csv"
+    upgrade_summary.extend(_get_upgrade_summary("thermal", thermal_upgrade_summary_file))
+
+    voltage_upgrade_summary_file = job_path / "VoltageUpgrades" / "voltage_summary.csv"
+    upgrade_summary.extend(_get_upgrade_summary("voltage", voltage_upgrade_summary_file))
+
+    return upgrade_summary
+
+
+def get_total_upgrade_costs_table(job_path, job_info):
+    total_upgrade_costs_file = job_path / "UpgradeCosts" / "total_upgrade_costs.csv"
+    if not total_upgrade_costs_file.exists():
+        return []
+    
+    try:
+        df = pd.read_csv(total_upgrade_costs_file)
+    except pd.errors.EmptyDataError as e:
+        logger.error("Failed to parse total upgrade costs file - '%s'", total_upgrade_costs_file)
+        logger.exception(e)
+        return []
+    
+    total_upgrade_costs = []
+    data = df.to_dict("records")
+    for item in data:
+        record = job_info._asdict()
+        record.update(item)
+        total_upgrade_costs.append(record)
+    return total_upgrade_costs
+
+
+def serialize_table(table, filename):
+    """Serialize a list of dictionaries to a CSV file."""
+    with open(filename, "w") as f:
+        assert table, filename
```

Comment:
Do we want to fail if there are no entries? Like if no jobs pass. Maybe we should create an empty file (just header) instead of asserting. I recognize that you followed what I did in make_summary_tables. I could make the changes.

Reply:
yeah, re-used the functions if u created.